libs/http2-manager/src/HTTP2/Client/Manager/Internal.hs

```diff
@@ -0,0 +1,337 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module HTTP2.Client.Manager.Internal where
+
+import Control.Concurrent
+import Control.Concurrent.Async
+import Control.Concurrent.STM
+import Control.Exception
+import Control.Monad
+import Control.Monad.IO.Class
+import Data.ByteString
+import qualified Data.ByteString as BS
+import Data.IORef
+import Data.Map
+import qualified Data.Map as Map
+import Data.Streaming.Network
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import Data.Unique
+import Foreign.Marshal.Alloc (mallocBytes)
+import qualified Network.HTTP2.Client as HTTP2
+import qualified Network.Socket as NS
+import qualified OpenSSL.Session as SSL
+import qualified System.TimeManager
+import Prelude
+
+data HTTP2Conn = HTTP2Conn
+  { backgroundThread :: Async (),
+    disconnect :: IO (),
+    -- See comment in 'startPersistentHTTP2Connection' about why this returns
+    -- '()'
+    sendRequestMVar :: MVar (Either Request ())
+  }
+
+type TLSEnabled = Bool
+
+type HostName = ByteString
+
+type Port = Int
+
+type Target = (TLSEnabled, HostName, Port)
```

Comment:
export this as well and use it more in the function signatures, instead of the 3 arguments?  i think currying is not very common here, and it'd be easier to read, i think.